misc/allow-block-list/src/lib.rs

```diff
@@ -231,23 +231,7 @@ where
         Ok(dummy::ConnectionHandler)
     }
 
-    fn on_swarm_event(&mut self, event: FromSwarm<Self::ConnectionHandler>) {
-        match event {
-            FromSwarm::ConnectionClosed(_) => {}
-            FromSwarm::ConnectionEstablished(_) => {}
-            FromSwarm::AddressChange(_) => {}
-            FromSwarm::DialFailure(_) => {}
-            FromSwarm::ListenFailure(_) => {}
-            FromSwarm::NewListener(_) => {}
-            FromSwarm::NewListenAddr(_) => {}
-            FromSwarm::ExpiredListenAddr(_) => {}
-            FromSwarm::ListenerError(_) => {}
-            FromSwarm::ListenerClosed(_) => {}
-            FromSwarm::NewExternalAddrCandidate(_) => {}
-            FromSwarm::ExternalAddrExpired(_) => {}
-            FromSwarm::ExternalAddrConfirmed(_) => {}
-        }
-    }
+    fn on_swarm_event(&mut self, _: FromSwarm<Self::ConnectionHandler>) {}
```

Comment:
Can we somehow make use of [`non_exhaustive_omitted_patterns_lint`](https://github.com/rust-lang/rust/issues/89554) here even though it is nightly only at this point? When we add a new `FromSwarm` variant, I find it useful to explicitly have to handle it in each of the `libp2p-xxx` crates in this monorepo. Given that each `libp2p-xxx` has to depend on the latest `libp2p-swarm` (due to workspace dependencies), this should not lead to any version mismatches.

Reply:
Yeah we definitely can. Just have to keep using nightly Rust in CI. We just need to add it to the `[workspace.lints]` table in the root `Cargo.toml`.

Reply:
I am in-different to be honest. I don't see a big gain in adopting this lint. It shouldn't block this PR as we can always make this change later as well.

Reply:
@mxinden Adopting this lint will require us to permanently lint with nightly in CI whereas at the moment, we will only lint with nightly until 1.75 is out.

I think for the moment, just doing a catch-all is simpler. We can change this at any point so I'd suggest to not block this PR on activating this lint but rather ship it as a follow-up :)

> I find it useful to explicitly have to handle it in each of the `libp2p-xxx` crates in this monorepo.

I'd like to challenge the usefulness of this. It might just be me but having to extend 30+ `match` statements with a new variant is a task where my brain shuts off instead of carefully thinking about each one whether or not I want to handle this variant here. When I introduce a new variant, there a few select places I want to handle it in and having to fix the entire workspace before I can have a clean `cargo clippy --all-targets --all-features` just blocks the flow of feature development.

To move forward with this (and unblock the `v0.53` release) we need to either add a catch-all variant to all these `match` blocks or remove them altogether. Without activating the lint, they are functionally equivalent and the latter is already implement here, hence I think deferring this is a) safe and b) faster.

Reply:
:+1: sounds good to me. Thanks for the additional reasoning.

Reply:
So does this lint just warn whenever there is a match on a non-exhaustive enum and not all branches are explicitly handled? The #[non_exhaustive] attribute just requires that any match has a catch-all match arm and then this lint forces a warning when that is the case to alert the developer that they should review it? I'm worried about consistent false positives causing developers to ignore warnings.

Reply:
I talked with @mxinden to get the answers to my questions. I was correct in my understanding. I think we should not add this simply because of the unbounded time we'll be dependent on the nightly toolchain for CI/CD.